### PR TITLE
Refactor IDPH ETL's

### PR DESCRIPTION
**Pull request by m0nhawk**
### Improvements
* Refactoring IDPH ETL's to make it simpler

### CODE REVIEW

File: covid19-etl-requirements.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
-boto3>=1.5<1.6
-requests>=2.23.0<3.0.0
+boto3>=1.5,<1.6
+requests>=2.23.0,<3.0.0
```
Comment (paulineribeyre) [inline covid19-etl-requirements.txt:2]: 😅

File: covid19-etl/helper/format_helper.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        str: generated "submitter_id"
+    """
+    submitter_id = "{}".format(node)
+    for v in args.values():
+        submitter_id += "_{}".format(v)
+
+    submitter_id = submitter_id.lower()
+    submitter_id = re.sub("[^a-z0-9-_]+", "-", submitter_id)
+
+    return submitter_id
```
Comment (paulineribeyre) [inline covid19-etl/helper/format_helper.py:24]:
```suggestion
    return submitter_id.strip("-")
```

File: covid19-etl/helper/format_helper.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    Resulting "submitter_id" only contains lowercase letters, digits, underscore and dash.
+
+    Args:
+        node (str): node name for "submitter_id"
+        args (dict): additional arguments to add to "submitter_id"
+
+    Returns:
+        str: generated "submitter_id"
+    """
+    submitter_id = "{}".format(node)
```
Comment (paulineribeyre) [inline covid19-etl/helper/format_helper.py:17]:
```suggestion
    submitter_id = node
```
? 😛

File: covid19-etl/etl/idph.py [hunk 4]
```diff
@@ -72,30 +61,28 @@ def files_to_submissions(self):
...
-        state = "IL"
 
         # they changed the URL on April 1, 2020
         if today > datetime.date(2020, 3, 31):
             url = "http://www.dph.illinois.gov/sitefiles/COVIDTestResults.json"
         else:
             url = f"https://www.dph.illinois.gov/sites/default/files/COVID19/COVID19CountyResults{today_str}.json"
-        self.parse_file(latest_submitted_date, state, url)
+        # self.parse_file(latest_submitted_date, state, url)
+        self.parse_file(today, url)
```
Comment (paulineribeyre) [inline covid19-etl/etl/idph.py:71]: why?

File: covid19-etl/helper/format_helper.py (hunk 2 above)
Comment (m0nhawk) [reply covid19-etl/helper/format_helper.py:24]: Does it mean `submitter_id` shouldn't have dashes or we want it to contain dashes?

File: covid19-etl/etl/idph.py (hunk 4 above)
Comment (m0nhawk) [reply covid19-etl/etl/idph.py]: Some testing code come through my 👀  when I was staging stuff.

File: covid19-etl/helper/format_helper.py (hunk 2 above)
Comment (paulineribeyre) [reply covid19-etl/helper/format_helper.py:24]: it's to remove the extra `-` at the beginning or at the end of the string

Comment (m0nhawk) [reply covid19-etl/helper/format_helper.py:24]: Ok, in the beginning of the string it shouldn't happen (it expects node name). But wouldn't bad to include this. 👍

Comment (paulineribeyre) [reply covid19-etl/helper/format_helper.py]: ah right. at the end it did happen when i was first working on the jhu etl

**Pull request merged**